src/modules/username.rs

```diff
@@ -0,0 +1,49 @@
+use ansi_term::{Color, Style};
+use std::env;
+use std::process::Command;
+
+use super::{Context, Module};
+
+/// Creates a segment with the current user's username
+///
+/// Will display the usename if any of the following criteria are met:
+///     - The current user isn't the same as the one that is logged in ($LOGNAME != $USER)
+///     - The current user is root (UID = 0)
+///     - The user is currently connected as an SSH session ($SSH_CONNECTION)
+pub fn segment(_context: &Context) -> Option<Module> {
+    let user = env::var("USER").ok();
+    let logname = env::var("LOGNAME").ok();
+    let ssh_connection = env::var("SSH_CONNECTION").ok();
+
+    let mut module_color = Color::Yellow.bold();
+
+    if user != logname || ssh_connection.is_some() || is_root(&mut module_color) {
+        let mut module = Module::new("username");
+        module.set_style(module_color);
+        module.new_segment("username", user?);
+
+        return Some(module);
+    }
+
+    None
+}
+
+fn get_uid() -> Option<u32> {
+    match Command::new("id").arg("-u").output() {
```

Comment:
It appears that `$UID` is an environment variable in bash and ZSH. Could we perhaps check for the environment variable and use `id -u` has a fallback if it doesn't exist?

Reply:
Ok, I'll try that and see how it goes